src/unitxt/stream_operators.py

```diff
@@ -102,6 +90,26 @@ def merge(self, multi_stream) -> List:
             left_on=self.left_on,
             right_on=self.right_on,
         )
+
+        def assert_col_values_are_identical(
+            df: pd.DataFrame, col_name_1: str, col_name_2
+        ):
+            assert df.apply(
+                lambda row: str(row[col_name_1]) == str(row[col_name_2]),
+                axis=1,
+            ).all()
+
+        common_cols_to_verify = ["data_classification_policy", "recipe_metadata"]
```

Comment:
Not sure I understand (also not documented), what happens when there are columns with same name in left and right?   Right now I think they are kept as separate values ,except "data_classification_policy" and "recipe_metadata" which must be the same.

Reply:
it will turn into "data_classification_policy_x" and "data_classification_policy_y" for example

Reply:
added a comment explaining it